src/labels_by_user_input.py

```diff
@@ -10,25 +13,65 @@
+You're trying to add/remove an unsupported label.
+Supported labels -
+{NEWLINE.join([f"/{key} to set {LABELS_DICT[key]}" for key in LABELS_DICT])}
+To remove a label use '-', for example: /-<LABEL_NAME>
+"""
+
+
+def labels_by_user_input(data, pull, commented_user):
     body = data["comment"]["body"]
-    last_commit = list(pull.get_commits())[-1]
-    if f"/{LABEL_VERIFIED}".lower() in body and LABEL_VERIFIED not in get_labels(
-        pull=pull
-    ):
-        add_label(pull=pull, label=LABEL_VERIFIED)
-        set_commit_status_success_verify(commit=last_commit)
-
-    if f"/un{LABEL_VERIFIED}".lower() in body:
-        remove_label(pull=pull, label=LABEL_VERIFIED)
-        set_commit_status_pending_no_verify(commit=last_commit)
-
-    if f"/{LABEL_APPROVE}".lower() in body and LABEL_APPROVE not in get_labels(
-        pull=pull
-    ):
-        add_label(pull=pull, label=LABEL_APPROVE)
-        set_commit_status_success_approve(commit=last_commit)
-
-    if f"/un{LABEL_APPROVE}".lower() in body:
-        remove_label(pull=pull, label=LABEL_APPROVE)
-        set_commit_status_pending_no_approve(commit=last_commit)
+    commented_user = data["comment"]["user"]["login"]
+    approver = commented_user in get_repo_approvers()
+    pr_labels = get_labels(pull=pull)
+    comment_labels = re.findall("(?:(?<=\\s)|(?<=^))/\\S*", body)
+
+    for label in comment_labels:
+        unlabel = label.startswith("/-")
+        # If user label doesn't start with '/-', only '/' is stripped
+        stripped_label = label.lower().lstrip("/-")
+
+        if stripped_label not in LABELS_DICT:
+
+            print(UNSUPPORTED_LABELS)
+            return pull.create_comment(
+                body=f"Hey @{commented_user},{UNSUPPORTED_LABELS}"
+            )
+
+        target_label = LABELS_DICT[stripped_label]
+        verified = LABEL_VERIFIED in target_label
+        approved = LABEL_APPROVE in target_label
+        last_commit = list(pull.get_commits())[-1]
+
+        if target_label in pr_labels and unlabel:
+
+            if approved:
+                set_commit_status_pending_no_approve(commit=last_commit)
+                target_label = list(
+                    (filter(lambda label: target_label in label, pr_labels))
+                )[
+                    0
+                ]  # Extract correct approve label
+
+            print(f"Removing {target_label} from {pull.title}")
+            remove_label(pull=pull, label=target_label)
+
+            if verified:
+                set_commit_status_pending_no_verify(commit=last_commit)
+
+        if target_label not in pr_labels and not unlabel:
+            if approved:
+                if not approver:
+                    continue
+                target_label = f"{commented_user}/{target_label}"
+                set_commit_status_success_approve(commit=last_commit)
+
+            print(f"Adding {target_label} to {pull.title}")
+            add_label(pull=pull, label=target_label)
```

Comment:
approved label should be set only if approver.

Reply:
Line 69 - if not approver, continue

Reply:
did I miss something?
line 74 is outside `if approved:`

Reply:
continue will cause the for loop to skip the rest of the code

Reply:
I think it's better to keep thing simple
Please check my PR and @sarahbx PR

Reply:
Your PR's implemented in this PR, and I think what @sarahbx did in her PR, is also mostly implemented in this pr, except for specific label mentions.
@sarahbx Correct me if I'm wrong, but lots of your PR's about implementing this PR.
'continue' relieves of writing the same thing multiple times

Reply:
@AdiZav looks like we have similar goals, just trying to keep it simple as I can. I've just split it up for @myakove, so please take another look, maybe we can use both.

Reply:
@AdiZav I think we should keep is simple, more readable and understood when we break it to labels